### ISSUE TITLE
a11y: Fix keyboard navigation, aria-label, role and focus display for top-left hamburger menu

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12769]
a11y Issues: 13, 7

The hamburger menu button (see below) does not show keyboard focus and does not response to enter/space to activate.

![image](https://github.com/user-attachments/assets/57c0fd75-277c-4b09-ad5d-d8f7105cc2aa)

This will require manual testing to check the focus presentation.
We could automate the testing of keyboard events to hide/show the expanded app bar.

Issue 7 - the icon is missing `area-label-by` (and probably also `role`=`button`)

To test:

- Load the app and login
- Press the Tab key
- Verify that hamburger icon shows the focus state
- Press enter - the app bar should expand
- Press enter - the app bar should collapse
- Press space - the app bar should expand
- Press space - the app bar should collapse
- Press SHIFT + Tab - Focus should be removed
- Press Tab - focus should be shown

**Comment by IsaSih** [comment]
Keyboard navigation, as well as the aria-label and role of the burger menu, working as expected in 
Rancher: v2.11-e9f0646414809cbaee8a1de474d8da152b522f06-head
Dashboard: master 938b866
Rancher CLI: v2.11.0-rc.1

<img width="950" alt="Image" src="https://github.com/user-attachments/assets/7416252d-58ae-4d8e-80a4-8e7d5773b8a5" />

**Comment by IsaSih** [comment]
Automated test coverage being addressed by PR https://github.com/rancher/dashboard/pull/12900 tracked by https://github.com/rancher/dashboard/issues/8986
